### PR TITLE
[docs] further clarifications RE: Hermes and remote debugging

### PR DESCRIPTION
**Pull request by keith-kurak**
# Why
I received a question linking to this section of the docs that expressed a concern that debugging was partially non-functional in Hermes.

Earlier in the doc, it clarifies that Hermes apps use the new JS Inspector, but if you zoom right down to this part, and you're someone who didn't really understand the difference between "debugging" and "remote debugging", I could see misunderstanding this section, thinking that your long-favored debugging technique no longer worked and that you had to use a particular protocol in order to debug the right way, but you weren't really sure how because it didn't say right there how to use said protocol.**

** It's me. Hi. I had to read this at least 8 times and then scroll up and down a few times to find the part that relieved my concerns that I was debugging wrong.

# How
Further differentiated remote vs new debugging, clarified that this is all informational, partially repeating that you'll automatically be debugging the right way if you choose to debug your Hermes app.

Also removed the link back to the Reanimated docs explaining why remote debugging does not work, because that page no longer has anything to say about remote debugging.

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/guides/using-hermes.mdx [hunk 1]
```diff
@@ -108,6 +108,6 @@ Alternatively, you can use the JavaScript inspector from the following tools:
 
 ### Can I use Remote Debugging with Hermes?
 
-One of the many limitations of [remote debugging](/more/glossary-of-terms/#remote-debugging) is that it does not work with modules built on top of [JSI](https://github.com/react-native-community/discussions-and-proposals/issues/91). For example, if your app uses [react-native-reanimated](https://github.com/software-mansion/react-native-reanimated) version 2 or higher, [remote debugging will not work](https://docs.swmansion.com/react-native-reanimated/docs/#known-problems-and-limitations).
+One of the many limitations of [remote debugging](/more/glossary-of-terms/#remote-debugging) is that it does not work with modules built on top of [JSI](https://github.com/react-native-community/discussions-and-proposals/issues/91), such as [react-native-reanimated](https://github.com/software-mansion/react-native-reanimated) version 2 or higher.
```
Comment (amandeepmittal) [inline docs/pages/guides/using-hermes.mdx:111]: Do you think its appropriate here to link to this Reanimated docs link: https://docs.swmansion.com/react-native-reanimated/docs/guide/debugging/?

File: docs/pages/guides/using-hermes.mdx [hunk 2]
```diff
@@ -108,6 +108,6 @@ Alternatively, you can use the JavaScript inspector from the following tools:
...
 
-Hermes supports [Chrome DevTools Protocol](https://chromedevtools.github.io/devtools-protocol/v8/) to debug JavaScript in place by connecting to the running engine on the device. This enables you to debug your app even when using JSI modules. This debugging technique is superior to remote debugging and replaces it for apps using Hermes.
+Hermes supports [Chrome DevTools Protocol](https://chromedevtools.github.io/devtools-protocol/v8/) to debug JavaScript in place by connecting to the  engine running on the device, as opposed to remote debugging, which executes JavaScript within a desktop Chrome tab. Hermes apps use this debugging technique automatically when you open the debugger in Expo Go or a development build.
```
Comment (amandeepmittal) [inline docs/pages/guides/using-hermes.mdx:113]:
```suggestion
Hermes supports [Chrome DevTools Protocol](https://chromedevtools.github.io/devtools-protocol/v8/) to debug JavaScript in place by connecting to the engine running on the device, as opposed to remote debugging, which executes JavaScript within a desktop Chrome tab. Hermes apps use this debugging technique automatically when you open the debugger in Expo Go or a development build.
```

Removing extra space.

**Review by amandeepmittal — APPROVED** [review]
Looks good to me 👍 Thanks @keith-kurak!

One small suggestion and we're good to ship these changes 🚢

File: docs/pages/guides/using-hermes.mdx (hunk 1 above)
Comment (keith-kurak) [reply docs/pages/guides/using-hermes.mdx:111]: I dunno, it's a good document, but Reanimated is mostly just serving as an example here, it's not really about debugging reanimated, per se.

Comment (amandeepmittal) [reply docs/pages/guides/using-hermes.mdx:111]: Got it!

**Pull request merged**